hamocc/mo_netcdf_bgcrw.F90

```diff
@@ -378,7 +378,7 @@ subroutine write_netcdf_var(ncid,desc,arr,klev,time)
       !   stop '(write_pnetcdf_var)'
       ! endif
 
-      deallocate(istart,icount,arr_g1)
+      deallocate(istart,icount,arr_l,arr_g1)
```

Comment:
Hi @TomasTorsvik , I am not sure, if I agree here to put in the `arr_l`, since from Fortran, `deallocate` seem to expect the object to be allocated, if I am not mistaken (and you deallocate it above, if I see it correct).

Reply:
The allocation of `arr_l` is different depending on IOTYPE==0/1, and the allocation is placed inside an `if ... else if ... endif` block. Hence the scope of the allocation should not reach outside this if-block, so there is a `deallocate`for each of them.

Because the `PNETCDF` CPP flag determines if the `IOTYPE==1` block is populated, there is a potential that the part after `else if`is empty. Hence we could introduce an error of the deallocation of `arr_l` is pulled outside the scope of the if-block in this case.